php/libraries/CandidateData.class.inc

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types=1);
+/**
+ * A CandidateData object is a DTO (data transfer object) to
+ * store a representation of data associated with a Candidate
+ * object in LORIS.
+ *
+ * The CandidateData may be partially loaded, in which case any
+ * getter trying to access an unknown will throw an exception.
+ * This allows modules to only load relevant data in a query,
+ * and avoid the overhead of the Candidate::singleton() constructor.
+ *
+ * For instance, a query that selected CenterID and ProjectID and
+ * timepoint related information:
+ * the database could instantiate a Candidate as:
+ *    $visit = new Candidate(new CandidateData($projectid, $centerid, $timepoints));
+ *
+ * with data loaded from the query, and then call
+ * `$candidate->isAccessibleBy($user)` without having to load all the
+ * other data that Candidate::singleton() does. If an attempt is
+ * made to access a property not in the DTO, an exception is thrown,
+ * rather than silently returning the incorrect value.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class CandidateData
+{
+    public \ProjectID $registrationProjectID;
+    public \CenterID $registrationCenterID;
+
+    public array $timepoints;
```

Comment:
I know this is not the endproduct but I think this object should at least have a candID. readonly probably

Reply:
I added a CandID but not readonly because that would up the minimum PHP version to 8.1.. also if that was readonly I think all of them should be.